src/AggregateFunctions/AggregateFunctionOrderBy.h

```diff
@@ -0,0 +1,186 @@
+                }
+            }
+            if (symb == ',')
+                if (!flag) {
+                    bitmap.push_back(true);
+                    flag = !flag;
+                }
+            if (isalpha(symb))
+                continue;
+            if (isspace(symb)) {
+                flag = !flag;
+            }
+        }
+        number_of_arguments_for_sorting = bitmap.size();
+    }
+
+    std::vector<std::pair<size_t, size_t>> findEqualRanges(size_t curr_idx) { 
+        std::vector<std::pair<size_t, size_t>> ans;
+        ans.emplace_back(0, data[0]->size()); 
+        for (size_t i = 0; i != curr_idx; ++i) {
+            if (ans.empty()) {
+                return ans;
+            }
+            std::vector<std::pair<size_t, size_t>> tmp;
+            while (!ans.empty()) {
+                auto range = ans.back(); 
+                ans.pop_back();
+                flag = false;
+                size_t start;
+                size_t counter = 0;
+                Field prev = *(data[i])[range.first];
+                Field curr;
+                for (size_t j = range.first + 1, j <= range.second; ++j) {
+                    if (data[i]->get(j, curr) == prev) {
+                        counter += 1;
+                        if (!flag) {
+                            flag = true;
+                            start = j - 1;
+                        }
+                    } else {
+                        if (flag) {
+                            flag = false;
+                            tmp.emplace_back(start, start + counter);
+                            counter = 0;
+                        }
+                    }
+                }
+            }
+            ans = tmp;
+        }
+        return ans;
+    }
+
+public:
+	AggregateFunctionOrderBy(AggregateFunctionPtr nested, const DataTypes & types, const Array & params_)
+        : IAggregateFunctionHelper<AggregateFunctionOrderBy>(types, params_)
+        , nested_func(nested), num_arguments(types.size())
+    {
+        if (params_.size() == 0)
+            throw Exception("Aggregate function " + getName() + " require at least one parameter", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+        if (params_[0].getType() != Field::Types::Which::String)
+            throw Exception("First parameter for aggregate function " + getName() + " must be String", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        parseString(params_[0].get());
+        if (num_arguments < number_of_arguments_for_sorting)
+            throw Exception("Aggregate function " + getName() + " require at several arguments", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+    }
+
+	String getName() const override
+    {
+        return nested_func->getName() + "OrderBy";
+    }
+
+    DataTypePtr getReturnType() const override
+    {
+        return nested_func->getReturnType();
+    }
+
+    void create(AggregateDataPtr place) const override
+    {
+        nested_func->create(place);
+    }
```

Comment:
Here you will need to create data for `FunctionOrderBy` itself and for nested function at the offset of `sizeof(FunctionOrderByData)`.